x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -149,21 +160,24 @@ export function getMigrations(
   const migrationRules820 = createEsoMigration(
     encryptedSavedObjects,
     (doc: SavedObjectUnsanitizedDoc<RawRule>): doc is SavedObjectUnsanitizedDoc<RawRule> => true,
-    pipeMigrations(addMappedParams)
+    pipeMigrations(addMappedParams, addSearchType)
   );
 
-  return {
-    '7.10.0': executeMigrationWithErrorHandling(migrationWhenRBACWasIntroduced, '7.10.0'),
-    '7.11.0': executeMigrationWithErrorHandling(migrationAlertUpdatedAtAndNotifyWhen, '7.11.0'),
-    '7.11.2': executeMigrationWithErrorHandling(migrationActions7112, '7.11.2'),
-    '7.13.0': executeMigrationWithErrorHandling(migrationSecurityRules713, '7.13.0'),
-    '7.14.1': executeMigrationWithErrorHandling(migrationSecurityRules714, '7.14.1'),
-    '7.15.0': executeMigrationWithErrorHandling(migrationSecurityRules715, '7.15.0'),
-    '7.16.0': executeMigrationWithErrorHandling(migrateRules716, '7.16.0'),
-    '8.0.0': executeMigrationWithErrorHandling(migrationRules800, '8.0.0'),
-    '8.0.1': executeMigrationWithErrorHandling(migrationRules801, '8.0.1'),
-    '8.2.0': executeMigrationWithErrorHandling(migrationRules820, '8.2.0'),
-  };
+  return mergeSavedObjectMigrationMaps(
+    {
+      '7.10.0': executeMigrationWithErrorHandling(migrationWhenRBACWasIntroduced, '7.10.0'),
+      '7.11.0': executeMigrationWithErrorHandling(migrationAlertUpdatedAtAndNotifyWhen, '7.11.0'),
+      '7.11.2': executeMigrationWithErrorHandling(migrationActions7112, '7.11.2'),
+      '7.13.0': executeMigrationWithErrorHandling(migrationSecurityRules713, '7.13.0'),
+      '7.14.1': executeMigrationWithErrorHandling(migrationSecurityRules714, '7.14.1'),
+      '7.15.0': executeMigrationWithErrorHandling(migrationSecurityRules715, '7.15.0'),
+      '7.16.0': executeMigrationWithErrorHandling(migrateRules716, '7.16.0'),
+      '8.0.0': executeMigrationWithErrorHandling(migrationRules800, '8.0.0'),
+      '8.0.1': executeMigrationWithErrorHandling(migrationRules801, '8.0.1'),
+      '8.2.0': executeMigrationWithErrorHandling(migrationRules820, '8.2.0'),
+    },
+    getEsQueryAlertSearchSourceMigrations(searchSourceMigrations)
```

Comment:
Is there a reason why `getEsQueryAlertSearchSourceMigrations` isn't injecting its migration under `8.2.0`? 🤔 
Am I missing something?

Reply:
Since we introduced `Search source` for `Es query` alert in `8.3.0`, we need migrations starting from that version. Any further modification of search source should be taken into consideration.

Reply:
Unless I'm missing something 🤔 I see @kertal gave a 👍 ?

Reply:
Maybe I'm missing something? :) getEsQueryAlertSearchSourceMigrations is building it's own migration map, that's currently empty, but once there are migrations they are merged with the other migrations

Reply:
dear @pmuellr / @mikecote , could you have a look? many thx!

Reply:
I understand why we need to do this. Every time the search source schema changes, we need to apply those changes to each ES Query rule params that are using search source. The `getEsQueryAlertSearchSourceMigrations` will generate migrations for future needs and `mergeSavedObjectMigrationMaps` will merge those with our own migrations explicitly defined.

I think it would be good to narrow `getEsQueryAlertSearchSourceMigrations` to 8.2+ or 8.3+ so it doesn't modify our existing migrations?

Reply:
oh, that's a good explanation, thanks Mike.
As this is a unique case - we should make this more explicit in the code.

At minimum - with a detailed comment.

IS this the sort of thing that the persistence toolkit could make easier or more maintainable? cc @stacey-gammon

Reply:
+1 for 

> I understand why we need to do this. Every time the search source schema changes, we need to apply those changes to each ES Query rule params that are using search source. The `getEsQueryAlertSearchSourceMigrations` will generate migrations for future needs and `mergeSavedObjectMigrationMaps` will merge those with our own migrations explicitly defined.
> 
> I think it would be good to narrow `getEsQueryAlertSearchSourceMigrations` to 8.2+ or 8.3+ so it doesn't modify our existing migrations?

yes, I think this is a good idea, also adding a comment